src/fidesops/migrations/versions/7abe778b7082_update_fideslang_data_categories.py

```diff
@@ -0,0 +1,217 @@
+    "user.provided.identifiable.government_id.national_identification_number": "user.government_id.national_identification_number",
+    "user.provided.identifiable.government_id.passport_number": "user.government_id.passport_number",
+    "user.provided.identifiable.health_and_medical": "user.health_and_medical",
+    "user.provided.identifiable.job_title": "user.job_title",
+    "user.provided.identifiable.name": "user.name",
+    "user.provided.identifiable.non_specific_age": "user.non_specific_age",
+    "user.provided.identifiable.political_opinion": "user.political_opinion",
+    "user.provided.identifiable.race": "user.race",
+    "user.provided.identifiable.religious_belief": "user.religious_belief",
+    "user.provided.identifiable.sexual_orientation": "user.sexual_orientation",
+    "user.provided.identifiable.workplace": "user.provided.identifiable.workplace",
+}
+
+DATA_MAP_DOWNGRADE = {v: k for k, v in DATA_MAP_UPGRADE.items()}
+
+
+def is_removed_downgrade(privacy_key: str) -> bool:
+    removed = (
+        "user.contact.address",
+        "user.contact.address.number",
+        "user.financial.payment_information",
+        "user.name.first",
+        "user.name.last",
+    )
+
+    return privacy_key in removed
+
+
+def is_removed_upgrade(privacy_key: str) -> bool:
+    removed = (
+        "account",
+        "account.contact",
+        "account.contact.city",
+        "account.contact.country",
+        "account.contact.email",
+        "account.contact.phone_number",
+        "account.contact.postal_code",
+        "account.contact.state",
+        "account.contact.street",
+        "account.payment.financial_account_number",
+        "user.derived",
+        "user.derived.identifiable",
+        "user.derived.nonidentifiable",
+        "user.derived.nonidentifiable.sensor",
+        "user.provided",
+        "user.provided.identifiable",
+        "user.provided.nonidentifiable",
+    )
+
+    return privacy_key in removed
+
+
+def update_field(field: dict, up_down: str) -> dict:
+    data_categories = field.get("data_categories")
+    updated = []
+    if up_down == "up":
+        if data_categories:
+            for data_category in data_categories:
+                if is_removed_upgrade(data_category):
+                    logger.info(
+                        "Removing %s, this is no longer a valid category", data_category
+                    )
+                if data_category:
+                    if not is_removed_upgrade(data_category):
+                        new = DATA_MAP_UPGRADE.get(data_category)
+                        if new:
+                            updated.append(new)
+                        else:
+                            updated.append(data_category)
+                    else:
+                        updated.append(data_category)
+        field["data_categories"] = updated
+
+        if field.get("fields"):
```

Comment:
This catches nested fields right? Since we're calling this in both `up_down` blocks, we could call this only once at the end outside of the `if` / `else` blocks.